src/main/java/frc/robot/FieldConstants.java

```diff
@@ -48,8 +48,8 @@ public static Pose2d flipPose(Pose2d pose) {
             Rotation2d rot = pose.getRotation();
             // reflect the pose over center line, flip both the X and the rotation
             return new Pose2d(FIELD_LENGTH - pose.getX(), pose.getY(), new Rotation2d(-rot.getCos(), rot.getSin()));
+        }else{        
+            return pose;
```

Comment:
This change is not useful; the "else" does not change the code flow. The point is that you are not checking if getAlliance returns NULL. It will crash in that case.